src/els_code_action_provider.erl

```diff
@@ -0,0 +1,183 @@
+-module(els_code_action_provider).
+
+-behaviour(els_provider).
+
+-export([ handle_request/2
+        , is_enabled/0
+        , options/0
+        ]).
+
+-include("erlang_ls.hrl").
+
+%%==============================================================================
+%% els_provider functions
+%%==============================================================================
+
+-spec is_enabled() -> boolean().
+is_enabled() -> true.
+
+%% /**
+%%  * The server provides code actions. The `CodeActionOptions` return type is
+%%  * only valid if the client signals code action literal support via the
+%%  * property `textDocument.codeAction.codeActionLiteralSupport`.
+%%  */
+%% codeActionProvider?: boolean | CodeActionOptions;
+
+-spec options() -> boolean() | map().
+options() ->
+  Capabilities = els_config:get(capabilities),
+  lager:info("code_actions: [Capabilities=~p]", [Capabilities]),
+  true.
+
+-spec handle_request(any(), els_provider:state()) ->
+  {any(), els_provider:state()}.
+handle_request({document_codeaction, Params}, State) ->
+  #{ <<"textDocument">> := #{ <<"uri">> := Uri}
+   , <<"range">>   := RangeLSP
+   , <<"context">> := Context } = Params,
+  Result = code_actions(Uri, RangeLSP, Context),
+  {Result, State}.
+
+%%==============================================================================
+%% Internal Functions
+%%==============================================================================
+
+
+%% result: (Command | CodeAction)[] | null where CodeAction is defined
```

Comment:
Same here. Please use `@edoc`. The specs should be used for documenting the types.